canu/generate/switch/config/config.py

```diff
@@ -1129,6 +1211,15 @@ def preserve_port(
             return port["lag"]
 
 
+def get_vlan_id(destination_rack_int, sls_variables, cabinet_key):
+    """Get the VLAN based on the cabinet."""
+    for cabinets in sls_variables[cabinet_key]:
```

Comment:
if cabinet_key isn't in sls_variables this will totally blow up.  Maybe calling code doesn't allow this to happen, but best would be to catch the exception and show an error.